torch/csrc/jit/passes/freeze_module.cpp

```diff
@@ -0,0 +1,395 @@
+#include <torch/csrc/jit/jit_log.h>
+#include <torch/csrc/jit/passes/freeze_module.h>
+
+#include <torch/csrc/jit/graph_executor_impl.h>
+#include <torch/csrc/jit/passes/alias_analysis.h>
+#include <torch/csrc/jit/passes/inliner.h>
+
+#include <stack>
+
+namespace torch {
+namespace jit {
+
+namespace {
+
+class AttributePropagator {
+ public:
+  AttributePropagator(script::Module& module) : module_(module) {}
+
+  void run(std::shared_ptr<Graph>& graph) {
+    Inline(*graph);
+    propagateAttributes(graph);
+    runOptimization(graph, /* unroll? */ false);
```

Comment:
I know quantization and potentially other parties wanted to run optimizations immediately after freezing. It might be worth removing the runOptimization call here to make freezing more composable.

Reply:
The optimization is important for the clean up. Getting rid of all unused attributes. I think it is better to return a clean graph and user can call whatever followup otimization post freezing.